app/controllers/question_answer/votes_controller.rb

```diff
@@ -142,12 +126,32 @@ def ensure_can_see_comment!(comment)
       @guardian.ensure_can_see!(comment.post)
     end
 
-    def ensure_can_vote(author_id)
-      if author_id == current_user.id
+    def ensure_can_vote(votable, votable_type)
```

Comment:
I think rather than having `votable_type` you can just do `votable.class.name` and check against that, I did it with bookmarks, that way there are no mismatches.

Reply:
Awesome suggestion, made it so. 👍🏼